src/php/Api/Infrastructure/PhelFnLoader.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Phel\Api\Infrastructure;
+
+use Phel\Lang\Collections\Map\PersistentMapInterface;
+use Phel\Lang\Registry;
+use Phel\Lang\TypeFactory;
+use Phel\Run\RunFacade;
+use Phel\Run\RunFacadeInterface;
+
+use function dirname;
+
+final class PhelFnLoader implements PhelFnLoaderInterface
+{
+    /** Prevent executing the RunCommand multiple times */
+    private static bool $phelInternalDocLoaded = false;
+
+    public function __construct(
+        private RunFacadeInterface $runFacade,
+    ) {
+    }
+
+    /**
+     * @return array<string,PersistentMapInterface>
+     */
+    public function getNormalizedPhelFunctions(): array
+    {
+        $this->loadAllPhelFunctions();
+
+        /** @var array<string,PersistentMapInterface> $normalizedData */
+        $normalizedData = [];
+        foreach ($this->getNamespaces() as $ns) {
+            $normalizedNs = str_replace('phel\\', '', $ns);
+            $moduleName = $normalizedNs === 'core' ? '' : $normalizedNs . '/';
+
+            foreach ($this->getDefinitionsInNamespace($ns) as $fnName => $fn) {
+                $fullFnName = $moduleName . $fnName;
+
+                $normalizedData[$fullFnName] = $this->getPhelMeta($ns, $fnName);
+            }
+        }
+        ksort($normalizedData);
+
+        return $normalizedData;
+    }
+
+    private function loadAllPhelFunctions(): void
+    {
+        if (self::$phelInternalDocLoaded) {
+            return;
+        }
+
+        $phelFile = __DIR__ . '/phel/doc.phel';
+        $this->runFacade = new RunFacade();
```

Comment:
We are injecting by constructor this `RunFacade`, I would say we don't need to initialize it here.

Reply:
Ups, I forgot to remove this line 😂